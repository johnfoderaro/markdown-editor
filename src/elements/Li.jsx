import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledLi = styled.li;

const Li = ({ children }) => (
  <StyledLi>
    {...children}
  </StyledLi>
);

Li.propTypes = { children: PropTypes.node.isRequired };

export default Li;
