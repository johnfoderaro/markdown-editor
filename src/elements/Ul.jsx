import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledUl = styled.ul`
  margin-left: .5rem;
  list-style: none;
`;

const Ul = ({ children }) => (
  <StyledUl>
    {children}
  </StyledUl>
);

Ul.propTypes = { children: PropTypes.arrayOf(PropTypes.element).isRequired };

export default Ul;
