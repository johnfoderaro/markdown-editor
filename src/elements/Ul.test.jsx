import React from 'react';
import { shallow } from 'enzyme';

import Ul from './Ul';
import Li from './Li';

let tree;

const content = ['a', 'b', 'c'];

beforeAll(() => {
  tree = shallow(
    <Ul>
      {content.map(c => <Li key={c}>{c}</Li>)}
    </Ul>,
  );
});

describe('<Ul />', () => {
  it('should be defined', () => {
    expect(tree).toBeDefined();
  });
  it('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
