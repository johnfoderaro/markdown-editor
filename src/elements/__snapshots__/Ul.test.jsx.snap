// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ul /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ul>
    <Li>
      a
    </Li>
    <Li>
      b
    </Li>
    <Li>
      c
    </Li>
  </Ul>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Li>
          a
        </Li>,
        <Li>
          b
        </Li>,
        <Li>
          c
        </Li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "a",
        "nodeType": "function",
        "props": Object {
          "children": "a",
        },
        "ref": null,
        "rendered": "a",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "b",
        "nodeType": "function",
        "props": Object {
          "children": "b",
        },
        "ref": null,
        "rendered": "b",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "c",
        "nodeType": "function",
        "props": Object {
          "children": "c",
        },
        "ref": null,
        "rendered": "c",
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  margin-left: .5rem;
  list-style: none;
",
        ],
      },
      "displayName": "styled.ul",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "ul",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Li>
            a
          </Li>,
          <Li>
            b
          </Li>,
          <Li>
            c
          </Li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "a",
          "nodeType": "function",
          "props": Object {
            "children": "a",
          },
          "ref": null,
          "rendered": "a",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "b",
          "nodeType": "function",
          "props": Object {
            "children": "b",
          },
          "ref": null,
          "rendered": "b",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "c",
          "nodeType": "function",
          "props": Object {
            "children": "c",
          },
          "ref": null,
          "rendered": "c",
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  margin-left: .5rem;
  list-style: none;
",
          ],
        },
        "displayName": "styled.ul",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "ul",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
